### ISSUE TITLE
arrow-body-style no braces for one-liner

### ISSUE THREAD

**Issue by mockdeep** [eslint/eslint#12924]
**What rule do you want to change?**

`arrow-body-style`

**Does this change cause the rule to produce more or fewer warnings?**

Mmm, the same? I guess it depends on the format of your existing code.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Probably a new option, something that gets across the idea of "always-multi-line-never-single-line".

**Please provide some example code that this change will affect:**

```js
// good
const valid_fields = fields.filter(field => field.is_valid);
const addresses = users.map((user) => {
  return user.address;
});

// bad
const valid_fields = fields.filter((field) => { return field.is_valid; });
```

**What does the rule currently do for this code?**

Depending on the configuration, it will require braces on one-liners or require that multi-line blocks don't have braces.

**What will the rule do after it's changed?**

Require that one-line arrow functions *don't* have braces and multi-line arrow functions *do* have braces.

**Are you willing to submit a pull request to implement this change?**

Willing, though it might be a while...

**Comment by kaicataldo** [comment]
What about a case where the arrow function gets split onto multiple lines? i.e.

```js
const valid_fields = fields.filter(field => 
    field.is_valid.heres_a_really_long_property_name);
```

**Comment by mockdeep** [comment]
@kaicataldo In this case I think I'd prefer to require the braces:

```ts
const valid_fields = fields.filter(field => {
  return field.is_valid.heres_a_really_long_property_name;
});
```

I think the no braces approach is tolerable in small doses, but can become confusing to interpret when it wraps.

**Comment by binoche9** [comment]
@kaicataldo Would you be willing to consider this idea? If I'm understanding correctly, the proposed behavior would match [arrow-return-shorthand from TSLint](https://palantir.github.io/tslint/rules/arrow-return-shorthand/), and would thus help those migrating from TSLint.

**Comment by kaicataldo** [comment]
The team has made the difficult decision to freeze stylistic rules (please see https://eslint.org/blog/2020/05/changes-to-rules-policies for more details), and this would fall under this category.

You might be able to enforce this by creating a custom rule using [eslint-rule-composer](https://www.npmjs.com/package/eslint-rule-composer).
